queue/index.js

```diff
@@ -0,0 +1,124 @@
+export class QueueChannel {
+  constructor(processing = false, data = []) {
+    this.processing = processing
+    this.data = data
+  }
+}
+
+export class Queue {
+  constructor(app, queues, key) {
+    this.app = app
+    this.queues = queues
+    this.key = key
+    this.clientId = undefined
+  }
+
+  setClientId(clientId) {
+    this.clientId = clientId
+    return this.clientId
+  }
+
+  add(action, meta) {
+    let queueKey
+    if (this.queues.size === 1) {
+      queueKey = 'main'
```

Comment:
`queueKey` should be `main` only if developer didn’t pass any queue name in `Server#type()`. It should not be dependent from number of defined queues.